1074 final tutorial qc

arfathpasha opened the pull request:
- cross compatible build and execution on past versions of docker, docker-compose, mac, linux
- dynamically generated ports and port config
- fixed bug in provision.py with assetstore permissions and missing packaging lib 
- made updates to some notebooks
- added tqdm to pip install

doori commented on docker/luna_tutorial/apply_ports.sh:
```diff
@@ -0,0 +1,56 @@
+
+unameOut="$(uname -s)"
+
+if [[ "$unameOut" == *"Darwin"* ]];
```
Nice improvement with host/port config. It would be good to add a comment in the nb that host/port is auto generated in the config files.

doori commented on docker/luna_tutorial/README.md:
```diff
@@ -8,16 +15,13 @@ $ make help [ARGS]
-
-    jupyterlab: http://localhost:8888  # NOTE insert token from vmount/logs/tutorial.log 
-    DSA:        http://localhost:8080
+In order to run this tutorial, first build the docker-compose images using `make build` and then lauch the containers using `make run`. Once the containers are launched, copy-paste the URL from the `make run` command into a browser and enter the token from `vmount/logs/tutorial.log`.  
```
We can probably remove `make build` as build is done when running `docker-compose up`.

doori commented on docker/luna_tutorial/README.md:
```diff
@@ -8,16 +15,13 @@ $ make help [ARGS]
-    DSA:        http://localhost:8080
+In order to run this tutorial, first build the docker-compose images using `make build` and then lauch the containers using `make run`. Once the containers are launched, copy-paste the URL from the `make run` command into a browser and enter the token from `vmount/logs/tutorial.log`.  
 
 You may then step through the notebooks in jupyterlab. Note that the url by default is set to localhost. If you need to run the tutorial from a remote system, you may specify the host IP and port of the remote system as arguments to the Makefile. 
```
no need to specify this anymore!

doori commented on docker/luna_tutorial/Makefile:
```diff
@@ -46,27 +46,36 @@ help:
+	mkdir -p vmount/logs vmount/db vmount/tmp vmount/assetstore
+	USER_ID=$(USER_ID) USER_NAME=$(USER_NAME) GROUP_NAME=$(GROUP_NAME) GROUP_ID=$(GROUP_ID) PWD=$(PWD) DOCKER_PATH=$(DOCKER_PATH) docker-compose up -d
+	@echo
+	@echo DSA will take a few minutes to become available at,
```
"DSA is being set up with sample images, annotations and admin user (admin/password)
It will take a couple minutes to become available at"

doori commented on docker/luna_tutorial/dockerfile_girder:
```diff
@@ -42,4 +43,4 @@ ENTRYPOINT ["/usr/bin/tini", "--"]
 
 RUN mkdir /logs && chmod -R 777 /logs
 USER $USER_NAME
-CMD python vmount/provision.py --sample-data && girder serve --database mongodb://mongodb:27017/girder
```
A comment on provision.py could be useful here:
- Creates an admin user (admin/password)
- Downloads data from https://data.kitware.com/#user/61b9f3dc4acac99f42ca7678/folder/61b9f4564acac99f42ca7692
and saves the slides at ~/vmount/PRO_12-123/data/toy_data_set
- Creates an assetstore at ~/vmount/assetstore
- Organizes the sample images in TCGA collection
- Uploads annotations to the slides.

doori reviewed (approved):
Nice improvement with dynamic host/port setting. Added minor suggestions mostly around documentation.

arfathpasha commented:
Thanks for reviewing Doori. I addressed all of your comments except the `make build` one. I have kept this target because I find it useful for debugging. Debugging with `make run` becomes a bit painful since it takes a few seconds to shut down containers before trying again, and if containers don't come up properly, which is often the case with debugging, then they have to be manually deleted.

aauker commented:
Re: failing tests, I see why, but I don't see any changes that'd cause them to fail, and the same tests are passing in dev.  I think either they were updated and this is an older branch, or something changed with the libraries versions that caused a changes to the resampling.  I recommend going ahead with the merge as is (if possible), and if things still fail (like a 50/50 chance) I can push a fix after.
